### ISSUE TITLE
Failed to initialize: ACTIVATION_ERROR

### ISSUE THREAD

**Issue by dariojavo** [Picovoice/rhino#425]
Hi,

Thanks for the repo. Currently, I have a code working with my corresponding key and context, which I can run from a bash file `code.sh`. So, when I execute it in CLI, everything runs normally. However, when I attempted to use it in a linux service `boot-rhino.service` for running it from boot, I receive an activation error. Please see below for details. 

### Expected behaviour

`access key=xxxx`
`context= xxx.rhn`

After running `bash code.sh` I receive a successful initialization with parameters `access_key` and `context` 

### Actual behaviour

After running `sudo systemctl start boot-rhino.service`  I received `Failed to initialize rhino with parameters \Parameters(access_key=xxxx, context=xxx.rhn): ACTIVATION_ERROR`

### Steps to reproduce the behaviour

1. Create a bash file that runs a script with initialization
2. Run that file (should work)
3. Create a service that load the bash file
4. Start the service
5. Check is rhino initialized, can use journalctl or something else.

Please let me know if more details are needed.

I'm not sure what is missing for be able to initialize from the systemctl, I check that services like wifi are enable while running my service.

Thanks in advance!

**Comment by mrrostam** [comment]
Hi there,

As a quick check, can you add something like `ping` to your bash script to make sure that you have access to the internet at that point? Also having access to the user home folder is essential, it would be helpful if there is a way to verify that.

**Comment by dariojavo** [comment]
Hi @mrrostam! thanks for your reply. Yes, I had verified that internet is working by using ping in the bash script. The script is also available for multi-user.

**Comment by mrrostam** [comment]
@dariojavo I used our [C file demo](https://github.com/Picovoice/rhino/tree/master/demo/c) and created a bash script and a new service as you described above, but couldn't reproduce the issue. I believe you already have `After=network.target` in the service config file, right?

**Comment by dariojavo** [comment]
Hi @mrrostam,

Thanks for your reply. Well, I'm interacting with ROS and the Rhino Speech-to-Intent engine. I was able to solve the issue, as suspected was a user privilege issue. I was able to solved it by specifying my user in the service.

 `[Unit]
After=multi-user.target, bluetooth.service

[Service]
User=<my_user>

[Install]
WantedBy=multi-user.target`

After that the activation key was successfully read! Later, and similar to before, I had some issues activating my Bluetooth device within the service, but I solved that issue  following [pulse audio for multiple-users](https://forums.gentoo.org/viewtopic-t-1110600-start-0.html), just in case someone will have the same issue :) 

Everything is working now! Thanks for the quick responses @mrrostam.

Best,
Dario.
